ipq807x: add support for TP-Link EAP620 HD v1

ssamsel opened the pull request:
Specifications:
  * SoC: Qualcomm IPQ8072A (64-bit Quad-core Arm Cortex-A53 @ 1.4 GHz)
  * Memory: 2x ESMT M15T4G16256A-DEBG2G (1 GiB DDR3-1866 13-13-13)
  * Serial Port: 3v3 TTL 115200n8
  * Wi-Fi: QCA5054 (4x4 5 GHz 802.11ax)
  * Wi-Fi: QCN5024 (2x2 2.4 GHz 802.11b/g/n/ax)
  * Ethernet: AR8031 (10/100/1000BASE-T)
  * Flash: Winbond W29N01HZSINF (128 MiB)
  * LEDs: 1x Blue Status (GPIO 42 Active High)
  * Buttons: 1x Reset (GPIO 50 Active Low)

Installation Instructions (Serial+TFTP):
  1. Solder 4 pin header to JP1 and bridge pads of R58 and R62.
  2. Connect 3V3 TTL port to TX, RX, and GND, which are positions 1, 2, and 3 respectively. Be sure to crossover TX and RX.
  3. Copy RAM firmware openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-initramfs-uImage.itb to a TFTP server's root that is in the same subnet as your AP.
  4. Power up the AP hold `Ctrl+B` in the serial console (115200n8) until autoboot is halted.
  5. Run the following commands in the U-boot prompt: 
  `# setenv serverip <TFTP server addr>`
  `# setenv ipaddr <addr of AP>`
  `# tftpboot 0x44000000 openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-initramfs-uImage.itb`
  `# bootm`
  You may need to type `Ctrl+C` and Enter before running these commands to clear invisible characters from the buffer.
  7. Run the following command in a terminal to copy the sysupgrade image to be installed (check IP address): 
   `$ scp -O openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-squashfs-sysupgrade.bin root@192.168.1.1:/tmp/`
  9. Activate the OpenWrt serial console and run the following commands: 
  `# cd /tmp` 
`# sysupgrade -n openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-squashfs-sysupgrade.bin`
  11. The AP will reboot and OpenWrt will be successfully installed.

Device support directly followed from EAP660HDv1 support 
Links: #15832

ssamsel committed:
ipq807x: add support for TP-Link EAP620 HD v1
Specifications:
  * SoC: Qualcomm IPQ8072A (64-bit Quad-core Arm Cortex-A53 @ 1.4 GHz)
  * Memory: 2x ESMT M15T4G16256A-DEBG2G (1 GiB DDR3-1866 13-13-13)
  * Serial Port: 3v3 TTL 115200n8
  * Wi-Fi: QCA5054 (4x4 5 GHz 802.11ax)
  * Wi-Fi: QCN5024 (2x2 2.4 GHz 802.11b/g/n/ax)
  * Ethernet: AR8031 (10/100/1000BASE-T)
  * Flash: Winbond W29N01HZSINF (128 MiB)
  * LEDs: 1x Blue Status (GPIO 42 Active High)
  * Buttons: 1x Reset (GPIO 50 Active Low)

Installation Instructions (Serial+TFTP):
  1. Solder 4 pin header to JP1 and bridge pads of R58 and R62.
  2. Connect 3V3 TTL port to TX, RX, and GND, which are positions 1, 2, and 3 respectively.
     Be sure to crossover TX and RX.
  3. Copy RAM firmware
     openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-initramfs-uImage.itb
     to a TFTP server's root that is in the same subnet as your AP.
  4. Power up the AP hold Ctrl+B in the serial console (115200n8) until autoboot is halted.
  5. Run the following commands in the U-boot prompt:
     # setenv serverip <TFTP server addr>
     # setenv ipaddr <addr of AP>
     # tftpboot 0x44000000 openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-initramfs-uImage.itb
     # bootm
     You may need to type Ctrl+C and Enter before running these commands
     to clear invisible characters from the buffer.
  6. Run the following command in a terminal to copy the sysupgrade image
     to be installed (check IP address):
     $ scp -O openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-squashfs-sysupgrade.bin root@192.168.1.1:/tmp/
  7. Activate the OpenWrt serial console and run the following commands:
     # cd /tmp
     # sysupgrade -n openwrt-qualcommax-ipq807x-tplink_eap620hd-v1-squashfs-sysupgrade.bin
  8. The AP will reboot and OpenWrt will be successfully installed.

Known Issues:
  * 5GHz radio instability (upstream current ath11k build bug maybe?)

Device support directly followed from EAP660HDv1 support
Links: #15832

Signed-off-by: Shymon Samsel <ssamsel@umass.edu>
Link: https://github.com/openwrt/openwrt/pull/17254
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8070-eap620hd-v1.dts:
```diff
@@ -0,0 +1,150 @@
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
```
```suggestion
#include "ipq8074-ac-cpu.dtsi"
```

If its IPQ8070A really, check SMEM ID for it

ssamsel replied:
Turns out it's actually an IPQ8072a, just like the EAP660HDv1

robimarko replied:
Even better, I see you already updated the commit description.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8070-eap620hd-v1.dts:
```diff
@@ -0,0 +1,150 @@
+	qcom,port_phyinfo {
+		port@6 {
+			phy_address = <4>;
+			port_id = <6>;
```
This looks weird, since I am not sure if QCA807x even does regular SGMII mode at all.

ssamsel replied:
I just went off what the stock firmware's device tree had, I can experiment with changing it if you thing this is incorrect

robimarko replied:
Well, the thing is that I dont see how this could actually work, especially since according to this its connected to UNIPHY2 via SGMII?

I have not seen QCA807x connected to UNIPHY1 and UNIPHY2 as they dont support QSGMII nor PSGMII which QCA807x uses.

Can you install `mdio-tools` and give me the output of:
`mdio 9* 0x5 0x1f`

Cause the only way I see this working is in a QSGMII+SGMII config where they are only using the SGMII link for the last port basically.

ssamsel replied:
```
root@OpenWrt:~# mdio 9*
 DEV      PHY-ID  LINK
0x04  0x004dd074  up
root@OpenWrt:~# mdio 9* 0x4
BMCR(0x00): 0x3100
  flags: -reset -loopback +aneg-enable -power-down -isolate -aneg-restart
         -collision-test
  speed: 100-full

BMSR(0x01): 0x796d
  capabilities: -100-t4 +100-tx-f +100-tx-h +10-t-f +10-t-h -100-t2-f -100-t2-h
  flags:        +ext-status +aneg-complete -remote-fault +aneg-capable +link
                -jabber +ext-register

ID(0x02/0x03): 0x004dd074

ESTATUS(0x0F): 0xa000
  capabilities: +1000-x-f -1000-x-h +1000-t-f -1000-t-h
root@OpenWrt:~# mdio 9* 0x4 0x1f
0x8111
root@OpenWrt:~# mdio 9* 0x5 0x1f
0xffff
root@OpenWrt:~#

robimarko replied:
Well, that is AR8031/3 PHY, that makes a lot more sense as that is an SGMII or RGMII PHY.

In that case, the whole qca807x package node is incorrect.

ssamsel replied:
Oops, my mistake

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-eap620hd-v1.dts:
```diff
@@ -0,0 +1,150 @@
+		reg = <0>;
+
+		qca8075_4: ethernet-phy@4 {
+			compatible = "ethernet-phy-ieee802.3-c22";
```
Considering that the PHY ID you captures shows that this is not QCA8075 but rather AR8031 or AR8033(They use the same ID) there should be no phy package node at all.

Just leave the PHY node and use the ar8031 label instead of qca8075_4

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-eap620hd-v1.dts:
```diff
@@ -0,0 +1,150 @@
+&switch {
+	status = "okay";
+	switch_lan_bmp = <ESS_PORT6>;
+	switch_mac_mode0 = <MAC_MODE_PSGMII>;
```
Drop this as this UNIPHY is clearly unused

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-eap620hd-v1.dts:
```diff
@@ -0,0 +1,150 @@
+	status = "okay";
+	switch_lan_bmp = <ESS_PORT6>;
+	switch_mac_mode0 = <MAC_MODE_PSGMII>;
+	switch_mac_mode1 = <MAC_MODE_SGMII_CHANNEL0>;
```
Drop this since port info shows that only UNIPHY2 is used

robimarko commented:
I have merged the BDF, so just add a commit that updates ipq-wifi, rest looks good to me

ssamsel commented:
> I have merged the BDF, so just add a commit that updates ipq-wifi, rest looks good to me

ipq-wifi should already be gtg

ssamsel commented:
> I have merged the BDF, so just add a commit that updates ipq-wifi, rest looks good to me

Done

robimarko commented:
Please make the ipq-wifi bump a separate commit, and use this script for generating it:
https://git.openwrt.org/?p=maintainer-tools.git;a=blob;f=update_git_source_package.sh

ssamsel committed:
ipq-wifi: update to Git HEAD (2024-12-18)

4b849214b137 ipq8074: add TP-Link EAP620 HD v1 BDF Add board file for TP-Link EAP620 HD v1. Sourced from bdwlan_US.bin in stock firmware 1.1.0 Build 20211028

Signed-off-by: Shymon Samsel <ssamsel@umass.edu>
Link: https://github.com/openwrt/openwrt/pull/17254
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Thanks! Rebased on top of main and merged!
